src/gui/socketapi/socketapi.cpp

```diff
@@ -296,15 +296,21 @@ void SocketApi::slotNewConnection()
 {
     // Note that on macOS this is not actually a line-based QIODevice, it's a SocketApiSocket which is our
     // custom message based macOS IPC.
-    QIODevice *socket = _localServer.nextPendingConnection();
+    SocketApiSocket *socket = _localServer.nextPendingConnection();
 
     if (!socket) {
         return;
     }
     qCInfo(lcSocketApi) << "New connection" << socket;
-    connect(socket, &QIODevice::readyRead, this, &SocketApi::slotReadSocket);
-    connect(socket, SIGNAL(disconnected()), this, SLOT(onLostConnection()));
-    connect(socket, &QObject::destroyed, this, &SocketApi::slotSocketDestroyed);
+    connect(socket, &SocketApiSocket::readyRead, this, &SocketApi::slotReadSocket);
+    connect(socket, &SocketApiSocket::disconnected, this, [socket] {
+        qCInfo(lcSocketApi) << "Lost connection " << socket;
+        // will trigger destroyed in the next execution of the main loop
```

Comment:
I would really appreciate a comment that indicates that this fixes a synchronization issue.

Reply:
the double remove was stupid in the first place